Blog post: Preparing the Terrain

oliverbarnes opened the pull request:
For successful consulting engagements. Just an outline for approval for now, I'm working on fleshing it out.

Please let me know of any other points worth touching on, within this topic

marcoow commented:
I mean there's not a whole lot there yet :) but I think the topic is interesting. I'd make sure to frame it for the right audience though. Instead of making this about what a consultancy should do, I'd say it should be about what we do and what our clients or companies working with consultancies in general can do to support making these projects successes.

beerinho commented on src/posts/2024-05-21-team-buy-in:
```diff
@@ -0,0 +1,56 @@
+
+## With team buy-in:
+
+Trust
```
One of the big things I've found here is that it's important to communicate the successes but also the failures. I find the balance of the two is great for building trust

beerinho reviewed (commented):
It would also be interesting to see some examples of good ways to get the client to buy-in

KevinBongart commented:
@oliverbarnes The overall topic and structure is good to me, the content too. However, because it's consulting jargon, it will all sound very abstract to the audience. We're "brought in to deliver business value" 😆, that's as abstract as it gets but going more specific and we're describing our day-to-day client work ("leading an architecture meeting to get DevOps buy-in on combining services into a monolith"… that's too specific for instance!).

IMO, what could invite readers in without feeling too abstract is leading sections with examples, not necessarily Mainmatter-related.

Starting the whole post with something like this:
> "Ugh, leadership is bringing in a bunch of consultants from Big Co to audit the team and identify inefficiencies across the team… this doesn't sound good, right?"
> Of course this will look like bad news to the client's team. Audit! Inefficiencies! Changes suggested by _outsiders_! The team is going to resist and prevent the implementation of positive changes. They would need a very good reason to be on board… and that's called _team buy-in_.

Even poking fun at consultants in general, and using the post as a way to highlight the care we put into helping the team find value in identifying and implementing change.

oliverbarnes commented:
> Starting the whole post with something like this:

totally agree, it'd be a lot more fun. we could even use an image of an incoming pirate ship 🏴‍☠️

oliverbarnes commented:
@KevinBongart I haven't got much creative juice these days, so I just added your own blurb to try it on for size, and polished some more.

emmasofiah2o reviewed (commented):
Just one small adjustment, but this is overall very well written :). I do agree that it sounds very technical, but I wouldn't necessarily say that's bad. We can discuss some ways to make it less formal tho, if you'd like.

marcoow reviewed (commented):
I'd be careful to not make this seem like it's mostly about what we need/want from the client (and what reads a bit like frustration about particular projects perhaps) but also/mainly about what we do to make projects collaborative and foster team buy-in, what the objectively valid reasons are for getting in outsiders for certain things rather than handling everything internally, etc.

oliverbarnes commented:
> Just one small adjustment, but this is overall very well written :). I do agree that it sounds very technical, but I wouldn't necessarily say that's bad. We can discuss some ways to make it less formal tho, if you'd like.

@emmasofiah2o thanks, yes I would :) I'm going to read with fresh yes as well, and try a more conversational tone.

> I'd be careful to not make this seem like it's mostly about what we need/want from the client (and what reads a bit like frustration about particular projects perhaps) but also/mainly about what we do to make projects collaborative and foster team buy-in

@marcoow yeah, the intention was for this to be a guide for managers on how to best bring us in. Without the frustration part, of course. Would it help it if it were less about what we need/want and more about a recommendation for best return over investment?

> what the objectively valid reasons are for getting in outsiders for certain things rather than handling everything internally, etc.

👍 I actually had this point noted for a separate blog post, but I can see how it can fit here. Will add it in

marcoow commented:
I think what I'm saying is I'd also add what we **do** to foster buy-in.

oliverbarnes commented:
Let me know if https://github.com/mainmatter/mainmatter.com/pull/2414/commits/696d5a5da9e2b8c52fee004b9372505004fbf896 and https://github.com/mainmatter/mainmatter.com/pull/2414/commits/092c1792466477b25d08533a5910a5b46ab7c585 help - I'm also pairing with Emma an hour from now to polish this together

emmasofiah2o commented on src/posts/2024-05-21-team-buy-in.md:
```diff
@@ -0,0 +1,109 @@
+Boutique, technical consultancies like Mainmatter are nothing like that, but still "consultant" is a title with plenty of loaded meanings and associations.
+
+As @KevinBongart puts it eloquently, the perception can be like:
+
```
@oliverbarnes this is how I'd change Kevin's quote: 

"Ugh, leadership is bringing in a bunch of consultants from Big Co to audit the team and point out our inefficiencies… this can't be good, right?"

In the team's view,  consultants are bad news: audits, inefficiencies, and changes suggested by _outsiders_! The natural reaction is to resist the implementation of all adjustments, even positive ones.
The team would need a very good reason to be on board. And that's called team buy-in.

oliverbarnes replied:
perfect - I've added it on acf90553e7d92c71f3b76f3ac058ed1765a55fe3

emmasofiah2o commented on src/posts/2024-05-21-team-buy-in.md:
```diff
@@ -0,0 +1,109 @@
+
+Boutique, technical consultancies like Mainmatter are nothing like that, but still "consultant" is a title with plenty of loaded meanings and associations.
+
+As @KevinBongart puts it eloquently, the perception can be like:
```
```suggestion
As our Engineering Manager, [Kevin Bongart](https://github.com/KevinBongart), puts it eloquently, the perception can be like:
```

emmasofiah2o commented on src/posts/2024-05-21-team-buy-in.md:
```diff
@@ -0,0 +1,109 @@
+
+Boutique, technical consultancies like Mainmatter are nothing like that, but still "consultant" is a title with plenty of loaded meanings and associations.
+
+As our Engineering Manager @KevinBongart puts it eloquently, the perception can be like:
```
Maybe we link to GitHub? So:
```suggestion
As our Engineering Manager [Kevin Bongart](https://github.com/KevinBongart) puts it eloquently, the perception can be like:
```

emmasofiah2o commented on src/posts/2024-05-21-team-buy-in.md:
```diff
@@ -0,0 +1,109 @@
+---
+title: "Preparing the Terrain"
```
How about: (although I am not sure the exact term is augmentation?)
```suggestion
title: "Team Buy-in: Preparing the Terrain for Augmentation"
```

oliverbarnes replied:
Yeah, I'm not sure - augmentation is part of what we do, there's assessment, strategic advice... perhaps simply consulting? I see what you mean about being clear what terrain we're actually talking about

emmasofiah2o replied:
We can also go in a different direction, like: "Paving the way for successful collaboration" (or engagement? What's more appropriate?). Either with "Team buy-in:" at the beginning or without.

emmasofiah2o replied:
Or if you like "Preparing the terrain" better we can also keep that of course

marcoow reviewed (commented):
I think this should be more about what we do which is:

* make clear we have an outside view, we understand the status quo is there for a reason but we don't know these reasons necessarily and can therefore ask questions more easily, also we weren't there during the history (doesn't mean we're not responsible but means we can have a clearer view that's not tainted by all the historic knowledge)
* we see more companies and teams than people who work at product companies for a long time; we allow clients to benefits from all these experiences and things that we see that work/don't work that they otherwise don't have access to
* when assessing/transforming, it's always super important to make clear that this is not a thread but an opportunity for everyone to make their (work) lives better which we foster by organizing workshops, team events, etc.

pablobm commented on src/posts/2024-05-21-team-buy-in.md:
```diff
@@ -0,0 +1,109 @@
+
+The earlier the intention to bring us in is communicated, the better. Before even making a decision, hearing the team out and addressing any concerns is a very good idea. Building consensus regarding the engagement. 
+
+Sometimes it's hard to get consensus, of course. But when dissenters have been heard in earnest, that goes a _long_ way towards getting people to disagree and commit.
```
```suggestion
Sometimes it's hard to get consensus, of course. But when dissenters have been heard in earnest, that goes a _long_ way towards getting people to [Disagree and Commit](https://en.wikipedia.org/wiki/Disagree_and_commit).
```

oliverbarnes commented on src/posts/2024-05-21-team-buy-in.md:
```diff
@@ -0,0 +1,109 @@
+
+We're complementing the team until they no longer need us. 
+
+### Alignment on scope
+
+An engagement can take many shapes and forms. It can have a tightly bounded scope, like updating the runtime and dependencies of a given codebase, or a much broader and open ended one, as with continuous assessment of team processes and practices. Sometimes implementation work might be done in tandem with continuous assessment and counselling, too.
+
+Whatever the scope, it needs to be clearly communicated as soon as it is known. People can become territorial over their work. Nobody likes someone looking over their shoulders nor receiving unsolicited advice. If there's a misunderstanding over the scope of our work, that's exactly how it'll feel to the client team.
```
I agree 100%, I don't think that's a good approach either. That's not what I'm trying to say, and I can try to word it differently if need be.

My point here specifically is about transparency of scope, specially in a low-trust environment. _How_ that's best "communicated" (not commanded), as I try to make clear overall in the blog post, is by discussing early and openly the intention and scope with the team, and then trying to get the buy-in. 

Tight scope:

"we're thinking of bringing in a team that can help with their Rust expertise, and help us migrate some Ruby services over to Rust. This can improve the stability and performance of our product, lower costs infrastructure costs etc. What do you [client team] think?) 

Broader engineering scope (which can obviously be complementary to the above, but should be clearly stated):

"we're thinking of bringing in a very experienced team that can help with engineering processes and practices, including in devops and architecture, team org and hiring. This can improve our overall productivity, speed up delivery, tighten our feedback loop, decrease team churn. What do you think?"

Also: 

"assessment" doesn't mean people need to listen to us, right. It just means we're looking at things and giving recommendations, which can be listened to or ignored. But to be able to assess we need access, and the team buy-in, again.

Seems like the sticking point might be the "continuous assessment" wording? If that's loaded, I can remove it altogether.

oliverbarnes commented on src/posts/2024-05-21-team-buy-in.md:
```diff
@@ -0,0 +1,109 @@
+
+Consultants get a bad rep sometimes, and often for good reason, specially due to powerful top tier consultancies involved in everything from privatization of public companies to disaster relief. They're everywhere these days.
+
+Boutique, technical consultancies like Mainmatter are nothing like that, but still "consultant" is a title with plenty of loaded meanings and associations.
```
Reworded to "Software engineering consultancies"

oliverbarnes commented:
> I think this should be more about what we do which is:
> 
> * make clear we have an outside view, we understand the status quo is there for a reason but we don't know these reasons necessarily and can therefore ask questions more easily, also we weren't there during the history (doesn't mean we're not responsible but means we can have a clearer view that's not tainted by all the historic knowledge)
> * we see more companies and teams than people who work at product companies for a long time; we allow clients to benefits from all these experiences and things that we see that work/don't work that they otherwise don't have access to
> * when assessing/transforming, it's always super important to make clear that this is not a thread but an opportunity for everyone to make their (work) lives better which we foster by organizing workshops, team events, etc.

Will incorporate

emmasofiah2o commented on src/posts/2024-05-21-team-buy-in.md:
```diff
@@ -0,0 +1,166 @@
+---
+title: "Preparing the Terrain for Sucessful Engagements"
```
```suggestion
title: "Preparing the Terrain for Successful Engagements"
```

emmasofiah2o commented:
> @emmasofiah2o I plan on addressing latest comments and wrapping up next week - I'll be on vacation and will have more free time then

Any news @oliverbarnes? :). Since tomorrow is my last official working day, I'm passing the release of this post on to @IdeliaRisella. 

@ Idelia: Oliver wanted to make some adjustments to this post, so it's not ready yet. Once he is done modifying it, make sure you get approval from someone on the team (ideally Marco or Kevin), change all file dates to the release date, and merge! I had also prepared a post in Notion, you can find it [here](https://www.notion.so/mainmatter/Blog-Post-Rotation-c008fb460107457aba52e8a98f683d35?pvs=4).

IdeliaRisella commented:
Hi @oliverbarnes , I am Idelia. I am taking over Emma's position in Marketing and would like to know if there's any update to this blog? I will be waiting for your cue to move on with this task. Thank you and nice to meet you here.

pablobm commented:
Quick update: Oliver is away on vacation. I think he's back tomorrow 🙂 (I should have commented last week!)

oliverbarnes commented:
Hi @IdeliaRisella 👋 yes, I'm back from vacation now and will continue on addressing the comments on the blog post. It's messaging isn't yet consensual (Marco wants some changes).

I'd also like to get @LukeMathWalker's and @SaeLLo's review on it, if they have the time to have a look before I wrap up.

SaeLLo commented on src/posts/2024-05-21-team-buy-in.md:
```diff
@@ -0,0 +1,166 @@
+
+Outsiders can be helpful, of course. External experience and a fresh view on
+things can be very useful. But, as with any collaboration, it's important to be
+aware of local context and history.
```
"given" maybe instead of "local"?

SaeLLo commented on src/posts/2024-05-21-team-buy-in.md:
```diff
@@ -0,0 +1,166 @@
+say when a team is expanding and re-organizing. And, as often happens, both
+things at the same time.
+
+We discuss that with leadership, and different stakeholders who might be
```
With the leadership team

SaeLLo commented on src/posts/2024-05-21-team-buy-in.md:
```diff
@@ -0,0 +1,166 @@
+
+## Happy path
+
+When the client team is bought-in, we get an environment of trust, a sweet spot
```
client's team

SaeLLo commented on src/posts/2024-05-21-team-buy-in.md:
```diff
@@ -0,0 +1,166 @@
+
+The earlier the intention to bring us in is communicated, the better. Before
+even making a decision, hearing the team out and addressing any concerns is a
+very good idea. Building consensus regarding the engagement.
```
...good idea - Building consensus...

IdeliaRisella commented:
Hi @oliverbarnes would you kindly check on this post again when you have time? 😊 It was scheduled to release this week but no worries, I'll push it to next week maybe. Please ping me when there's changes. Thank you 🙏🏼

oliverbarnes commented:
For sure, will do Idelia. Can't promise a release yet as there's no consensus on the messaging (there might be more rounds of comments), but I'd love to wrap up soon.

marcoow commented on src/posts/2024-05-21-team-buy-in.md:
```diff
@@ -0,0 +1,114 @@
+
+[Reach out](/contact/) to talk about how we can help your team.
+
+Find [further information](/startups/) on how we can help you to burst through
```
```suggestion
Find [further information](/services/team-reinforcement/) on how we can help you to burst through
```

works better here

marcoow reviewed (commented):
I think this is good now :+1:

oliverbarnes commented:
> let's get the build fixed so we can see the preview

I'll go through the build breakage, but I'm guessing the linting is broken removed the line breaks to comply with Chris' proposed new convention here: https://github.com/mainmatter/mainmatter.com/pull/2492

Could we get that merged so I can rebase on top of it?

oliverbarnes commented:
Yeah, ran `prettier --write src/posts/2024-05-21-team-buy-in.md` and it added a bunch of line breaks

marcoow commented:
there's actual build errors – look at the log of any of the GitHub Actions jobs

LukeMathWalker commented on src/posts/2024-05-21-team-buy-in.md:
```diff
@@ -0,0 +1,114 @@
+Software engineering consultancies like Mainmatter are nothing like that, but still "consultant" is a title with plenty of loaded meanings and associations.
+
+Outsiders _can_ be helpful. External experience and a fresh view on things can be very useful.
+
```
I'd remove this newline here. It all fits together as a single paragraph.

LukeMathWalker commented on src/posts/2024-05-21-team-buy-in.md:
```diff
@@ -0,0 +1,114 @@
+
+And, as often happens, both at once.
+
+We discuss that with the leadership team and with different stakeholders who might be involved in bringing us in to prepare the terrain for the engagement.
```
I don't think this sentence is needed.

oliverbarnes replied:
I see what you mean, but Marco asked to emphasize what we actually do in practice, besides understanding context.

LukeMathWalker commented on src/posts/2024-05-21-team-buy-in.md:
```diff
@@ -0,0 +1,114 @@
+
+A virtuous cycle ensues, where implementing changes is progressively faster and more effective, and ideas flow freely back and forth. Together we shift into continous improvement mode. Everybody wins.
+
+So. Just how do you get buy-in?
```
```suggestion
So. How do you get buy-in?
```

LukeMathWalker commented on src/posts/2024-05-21-team-buy-in.md:
```diff
@@ -0,0 +1,114 @@
+
+## Get the team involved early on
+
+The earlier the intention to bring us in is communicated, the better. Before even making a decision, hearing the team out and addressing any concerns is a very good idea - building consensus towards the engagement.
```
```suggestion
The earlier the intention to bring us in is communicated, the better. Before even making a decision, hearing the team out and addressing any concerns is a very good idea - building trust in the engagement.
```

LukeMathWalker commented on src/posts/2024-05-21-team-buy-in.md:
```diff
@@ -0,0 +1,114 @@
+
+The earlier the intention to bring us in is communicated, the better. Before even making a decision, hearing the team out and addressing any concerns is a very good idea - building consensus towards the engagement.
+
+Sometimes it's hard to get consensus, of course. But when dissenters have been heard in earnest, that goes a _long_ way towards getting people to [disagree and commit](https://en.wikipedia.org/wiki/Disagree_and_commit).
```
```suggestion
Sometimes it's hard to get that trust, of course. But when dissenters have been heard in earnest, that goes a _long_ way towards getting people to [disagree and commit](https://en.wikipedia.org/wiki/Disagree_and_commit).
```

LukeMathWalker commented on src/posts/2024-05-21-team-buy-in.md:
```diff
@@ -0,0 +1,114 @@
+
+Some very successful engagements happened when we were invited by people from the tech or product teams, who are closer to where the action will happen. This can help a lot in getting buy-in.
+
+When the invitation comes from the very top, it makes a huge difference when management goes the extra mile and talks to the teams that will directly be involved in and impacted by the engagement.
```
I get what this paragraph wants to communicate, but I think it could be reworded to be easier to read/parse 🤔

oliverbarnes replied:
how about this: 1c495228cb5e73456be4dafbd5ff73cfe1847b83

oliverbarnes commented:
Most build jobs fixed, but I can't access Percy after signing up with google, and it's not immediately clear to me what needs to happen in the Gravity dashboard (maybe somebody else needs to approve it?).

I've updated the blog date to Monday, July 29th btw
